vendor/views/vendor_admin.py

```diff
@@ -183,4 +197,97 @@ def post(self, request):
             price.offer = offer
             price.save()
 
-        return redirect('vendor_admin:manager-offer-list')
\ No newline at end of file
+        return redirect('vendor_admin:manager-offer-list')
+
+
+class AdminSubscriptionListView(LoginRequiredMixin, ListView):
+    '''
+    List of all the invoices generated on the current site.
+    '''
+    template_name = "vendor/manage/receipt_list.html"
+    model = Receipt
+
+    def get_queryset(self):
+        return self.model.objects.filter(products__in=Product.on_site.all(), order_item__offer__terms__lt=TermType.PERPETUAL)
+
+
+class AdminSubscriptionDetailView(LoginRequiredMixin, DetailView):
+    '''
+    Gets all Customer Profile information for quick lookup and management
+    '''
+    template_name = 'vendor/manage/subscription_detail.html'
+    model = Receipt
+    slug_field = 'uuid'
+    slug_url_kwarg = 'uuid'
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        payment = Payment.objects.get(transaction=context['object'].transaction)
+
+        context['payment'] = payment
+        context['payment_form'] = CreditCardForm(initial={'payment_type': PaymentTypes.CREDIT_CARD})
+        context['billing_form'] = AddressForm(instance=payment.billing_address)
+
+        return context
+
+
+class AdminProfileListView(LoginRequiredMixin, ListView):
+    """
+    List of CustomerProfiles on site
+    """
+    template_name = "vendor/manage/profile_list.html"
+    model = CustomerProfile
+    queryset = CustomerProfile.on_site.all()
+
+class AdminProfileDetailView(LoginRequiredMixin, DetailView):
+    '''
+    Gets all Customer Profile information for quick lookup and management
+    '''
+    template_name = 'vendor/manage/profile_detail.html'
+    model = CustomerProfile
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+
+        context['receipts'] = self.object.receipts.filter(products__in=Product.on_site.all(), order_item__offer__terms__gte=TermType.PERPETUAL)
+        context['subscriptions'] = self.object.receipts.filter(products__in=Product.on_site.all(), order_item__offer__terms__lt=TermType.PERPETUAL)
+
+        # context['products'] = [ product for receipt in self.object.receipts.all() for product in receipt.products.all() ]
+
+        return context
+
+
+class VoidProductView(LoginRequiredMixin, View):
+    success_url = reverse_lazy('vendor_admin:manage-profiles')
+
+    def post(self, request, *args, **kwargs):
+        receipt = Receipt.objects.get(uuid=self.kwargs["uuid"])
+        receipt.end_date = timezone.now()
+        receipt.save()
+
+        messages.info(request, _("Customer has no longer access to Product"))
+        return redirect(request.META.get('HTTP_REFERER', self.success_url))
+
```

Comment:
Nitpick: make sure to add an extra blank line between classes.

Reply:
Line Added

Reply:
👍